server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -887,7 +887,7 @@ public boolean equals(Object obj) {
                 && Objects.equals(seqNo, other.seqNo)
                 && Objects.equals(primaryTerm, other.primaryTerm)
                 && Objects.equals(source, other.source)
-                && Objects.equals(fields, other.fields)
+                && Objects.equals(getFields(), other.getFields())
```

Comment:
A possible fix for equals as a separate commit here, can remove if not appropriate or should it's own separate PR.
Sharing the same SearchHit structures in the eql resful API implementation, and the equality checks would fail without this  change in place.
Following the same thing that was done a line below with highlightFields accessors.

Reply:
It might be a bit cleaner to make this a separate PR to master and 7.x and then merge it in. In any case, I think this requires a modification to SearchHitTests that would trigger the issue that you are trying to fix here.

Reply:
Sounds good, will update this PR once if goes through the master.

Reply:
This file change is removed now from PR.